src/components/structures/MatrixChat.js

```diff
@@ -592,23 +575,40 @@ module.exports = React.createClass({
         this.scrollStateMap[roomId] = state;
     },
 
-    onLoggedIn: function(credentials) {
+    _doLogin(creds) {
+        Lifecycle.login(creds, {
+            syncTimelineLimit: this.props.config.sync_timeline_limit,
+        });
+    },
+
+    _onHaveCredentials: function(credentials) {
```

Comment:
`_onHaveCredentials` isn't terribly descriptive. how about `_handleLoginCredentials`?

in fact, is there any reason for it at all? Why not just use `Lifecyle.login`?